topostats/processing.py

```diff
@@ -384,9 +385,11 @@ def run_grainstats(
             LOGGER.info(f"[{filename}] : Calculated grainstats for {len(grainstats_df)} grains.")
             LOGGER.info(f"[{filename}] : Grainstats stage completed successfully.")
             return grainstats_df, height_profiles_dict, grainstats_calculator.grain_crops
-        except Exception:
-            LOGGER.info(
-                f"[{filename}] : Errors occurred whilst calculating grain statistics. Returning empty dataframe."
+        except Exception as e:
+            _, _, e_traceback = sys.exc_info()
+            e_line_no = e_traceback.tb_lineno
+            LOGGER.error(
+                f"[{filename}] : An error occurred whilst calculating grain statistics on line {e_line_no}: {e}\nReturning empty dataframe."
```

Comment:
The logger is setup in `topostats/logs/logs/py` and we do set `LOG_ERROR_FORMATTER` to include `%(lineno)s` so I'm not sure whether you need to import `sys` and get a traceback to include the line-number. It might just be sufficient to `LOGGER.info` > `LOGGER.error` and the line number would be reported.